### ISSUE TITLE
Cargo install fails with latest on Ubuntu 22.04

### ISSUE THREAD

**Issue by fquinner** [casey/just#1495]
Cargo build fails on default Ubuntu 22.04

The Cargo / rustc version is 1.61.0.

Fails when running:

```
cargo install just
```

The failure is:

```
error[E0015]: cannot call non-const fn `std::sync::Mutex::<interrupt_handler::InterruptHandler>::new` in statics
  --> /home/fquinn/.cargo/registry/src/github.com-1ecc6299db9ec823/just-1.10.0/src/interrupt_handler.rs:17:48
   |
17 |     static INSTANCE: Mutex<InterruptHandler> = Mutex::new(InterruptHandler::new());
   |                                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: calls in statics are limited to constant functions, tuple structs and tuple variants

For more information about this error, try `rustc --explain E0015`.
```

And it happens for both 1.10.0 and 1.11.0 (1.9.0 works fine) if specified explicitly on cargo command line so thinking it may be related to #1442

**Comment by benmoss** [comment]
Any reason you don't want to use Rust 1.63 or newer? Are you using the apt packages and they're still on 1.61?

**Comment by fquinner** [comment]
Yeah all apt based defaults trying to keep it vanilla. No reason other than default / out of the box defaults.

**Comment by benmoss** [comment]
I think you're out of luck, I'd use [rustup](https://rustup.rs/) to get a newer version or use a prebuilt binary

**Comment by casey** [comment]
I just merged #1496, which should make the error message better in the future. (@benmoss thanks for the PR!)
